Marlin/src/gcode/calibrate/G28.cpp

```diff
@@ -411,11 +411,18 @@ void GcodeSuite::G28() {
     TERN_(HOME_Z_FIRST, if (doZ) homeaxis(Z_AXIS));
 
     const bool seenR = parser.seenval('R');
-    const float z_homing_height = seenR ? parser.value_linear_units() : Z_HOMING_HEIGHT;
+    float z_homing_height;
+    if (seenR)
```

Comment:
Originally the `R` parameter was meant to simply be 0 for cases where a raise was not wanted, and was not really designed for the case used by some LCD implementations where it has been set to "5." The implementers of the code for those LCDs were not aware of this, and may have expected `R5` to raise 5mm above the current position after modifying Z (for something like a power-loss recovery). So we need to look at their custom handling, if they are not simply using the standard recovery.resume behavior.

In general, when first homing a machine after it has been booted, the initial raise always works because the current Z position is reset to `Z_IDLE_HEIGHT`, which is 0 for most machines, but may be a large value for machines whose beds are susceptible to gravity.